Standards/scs-0212-v1-requirements-for-container-registries.md

```diff
@@ -0,0 +1,124 @@
+One use case would be the offering of private registries for customers by CSPs, which means that a CSP could
+offer private container registries either as a service or as a provided "recipe" to deploy a private registry
+utilizing the CSPs infrastructure.
+Another use case would be the selection of a registry for the SCS reference implementation.
+
+The idea and purpose of this document is to specify what requirements a specific technical container
+registry implementation (i.e. software solution) needs to fulfill for an SCS-compliant registry.
+
+## Design considerations
+
+There are numerous features that should be evaluated for a container registry solution.
+It is important to assess the registries based on the requirements of the OSS health checks and
+desired features defined by the SCS. The following two subsections show these requirements.
+
+### OSS health check
+
+It is important to evaluate the health of a project before even evaluating the project for its feature set.
+A project should therefore fulfill all OSS health checks be defined by the
+[OSS-Health](https://github.com/SovereignCloudStack/standards/blob/main/Drafts/OSS-Health.md) document.
+This document evaluates the health of the open-source projects that were selected from
+the currently available solutions. The container registry software must fulfill all OSS
+health checks defined below:
+
+- Four Opens (code is fully open source, community is open and diverse, development process is open, design process is open)
+- Maturity
+- Security
+- Activity
+- Lock-in risk assessment
+
+### Required and desirable features check
+
+A container registry provides a specific feature set, which can be mapped out against the requirements
+described in this section. The registry should generally be robust (e.g. operate under heavy load) and
+secure in order to be acceptable for the SCS standard. Therefore, a required and optional feature set
+were defined, to which a container registry must abide and be evaluated against.
+
+#### Required features
+
+- Audit Logs
+  - ability to record use in auditable logs so that activity can be traced to a single user
+- Authentication of system identities
+  - support for authenticating system identities like Kubernetes clusters
+  - possibly support for dynamic identity tokens from some IdP
+- Authentication of users
+  - support for multiple authentication systems (IdM integration)
+  - user and user account management
+- Authorization
+  - role-based access control to ensure strict access controls
+- Automation
+  - integration with CI/CD tools e.g. via webhooks
+- Vulnerability scanning
+  - reveal security vulnerabilities in container images
+- Content Trust and Validation
+  - verify image authenticity before running
+  - image signing
+- Multi-tenancy
+  - container registry is able to serve multiple tenants (projects, teams, namespaces)
+  - implementation on the storage level, see e.g. [Keppel](https://github.com/sapcc/keppel#overview)
```

Comment:
Maybe summarize shortly why especially "Keppel" would be a good choice for Multi-tenancy storage level implementation.

Reply:
There was a nice discussion about registry multitenancy, see this [thread](https://github.com/SovereignCloudStack/standards/pull/212#discussion_r1104392438).

TLDR: Keppel uses multi-tenant-aware storage drivers so each customer gets their own separate storage space for blobs.

Reply:
Changed in https://github.com/SovereignCloudStack/standards/pull/458/commits/86355a4f9249b0c438a287d00bc160b1e7edd743